dcore/state.py

```diff
@@ -120,6 +121,10 @@ def dump(self):
             self.diagnostic_data = defaultdict(partial(defaultdict, list))
 
         if (next(self.dumpcount) % self.output.dumpfreq) == 0:
+
+            if 'Courant' in self.diagnostic_fields.diagnostic_field_dict.keys():
+                self.diagnostic_fields.Courant(self)
+
```

Comment:
Thanks @wence- that's a zillion times better!

________________________________
From: Lawrence Mitchell <notifications@github.com>
Sent: 04 May 2016 15:43:26
To: firedrakeproject/dcore
Cc: Shipton, Jemma; Author
Subject: Re: [firedrakeproject/dcore] Courant number (#34)


In dcore/state.py<https://github.com/firedrakeproject/dcore/pull/34#discussion_r62051195>:

> @@ -120,6 +121,10 @@ def dump(self):
>              self.diagnostic_data = defaultdict(partial(defaultdict, list))
>
>          if (next(self.dumpcount) % self.output.dumpfreq) == 0:
> +
> +            if 'Courant' in self.diagnostic_fields.diagnostic_field_dict.keys():
> +                self.diagnostic_fields.Courant(self)
> +


Now you can do:


diagnostic_fields = [CourantNumber(), SomeOtherDiagnostic()]

state = State(..., diagnostic_fields)

And then in your dump method:


def dump(self, ...):
    for diag in self.diagnostic_fields:
        to_dump.append(diag.compute())

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/firedrakeproject/dcore/pull/34/files/716034aade19eaf4c06ec17aa9ce9860fc487b2c#r62051195>